File: src/main/hardplayer/MainStrategy.java
package hardplayer;

import battlecode.common.RobotController;

class MainStrategy extends Static implements Strategy {

	public void execute(RobotController myRC) {
		while(true) {
			try {
				switch(myRC.getType()) {
					case ARCHON:
						new ArchonPlayer(myRC).run();
						break;
					case SOLDIER:
						new SoldierPlayer(myRC).run();
						break;
					case SCOUT:
						new ScoutPlayer(myRC).run();
						break;
					case SCORCHER:
						new ScorcherPlayer(myRC).run();
						break;
					default:
						debug_println("I don't know what kind of robot I am!");
						myRC.suicide();
				}
			} catch(Exception e) { debug_stackTrace(e); }
			myRC.yield();
		}
	
	}

}

File: src/main/hardplayer/MainStrategy.scala
package hardplayer

import battlecode.common.Chassis
import battlecode.common.ComponentType
import battlecode.common.RobotController

import hardplayer.Static._
import hardplayer.sensor._

object MainStrategy extends Strategy {

	override def execute(myRC : RobotController) {

		while(true) {
			try {
				var comp = myRC.components()
				myRC.getChassis() match {
					case Chassis.BUILDING => {
						for(c <- comp) {
							if(c==ComponentType.RECYCLER) {
								new MinePlayer(myRC).run()
							}
						}
					}
					case Chassis.LIGHT => {
						for(c <- comp) {
							if(c==ComponentType.CONSTRUCTOR) {
								new ConstructorPlayer(myRC).run()
							}
							else {
								new LarvaPlayer(myRC).run() 
							}
						}
					}
				}
			} catch {
				case e : Exception => { debug_stackTrace(e) }
			}
		}
	
	}

}
